extensions-core/azure-extensions/src/main/java/org/apache/druid/storage/azure/output/AzureStorageConnectorProvider.java

```diff
@@ -40,22 +44,24 @@ public class AzureStorageConnectorProvider extends AzureOutputConfig implements
   @Global
   @JacksonInject
   AzureStorage azureStorage;
+  @JacksonInject
+  Injector injector;
 
   @JsonCreator
   public AzureStorageConnectorProvider(
       @JsonProperty(value = "container", required = true) String container,
       @JsonProperty(value = "prefix", required = true) String prefix,
-      @JsonProperty(value = "tempDir", required = true) File tempDir,
       @JsonProperty(value = "chunkSize") @Nullable HumanReadableBytes chunkSize,
       @JsonProperty(value = "maxRetry") @Nullable Integer maxRetry
   )
   {
-    super(container, prefix, tempDir, chunkSize, maxRetry);
+    super(container, prefix, chunkSize, maxRetry);
   }
 
   @Override
   public StorageConnector get()
   {
-    return new AzureStorageConnector(this, azureStorage);
+    final File tempDir = injector.getInstance(Key.get(File.class, Names.named(DataSourceTaskIdHolder.TMP_DIR_BINDING)));
```

Comment:
Thanks so much for this change @adarshsanjeev :+1: 
Although, it might be worth checking if you can directly `JacksonInject` the File object - one way to try it could be to do : 
```
@JacksonInject
@Named(DataSourceTaskIdHolder.TMP_DIR_BINDING)
File tempDir;
```

Also, could consider adding the `JacksonInject` in the constructor itself to keep all the relevant things there.

Reply:
Lastly, could be worth having the temp dir instance setup for other processes as well, and also for indexers

Reply:
I tried the above initially, but injecting the tempDir as a variable fails (maybe since it is a jackson inject, and that binding has not been added?).

Reply:
does it fail for processes which are not peons? if that's the case you can try also marking this `Nullable` 
also, if you have the error available, please feel free to paste it here

Reply:
The above code fails for peons with the error
 ```Guice configuration errors: 1) No implementation for java.io.File annotated with @com.google.inject.name.Named(value="druidTempDirectory") was bound. while locating java.io.File annotated with @com.google.inject.name.Named(value="druidTempDirectory") 1 error```